EIPS/eip-abstract_wallet_transaction_api.md

```diff
@@ -0,0 +1,218 @@
+
+## Specification
+
+Two new JSON RPC methods are added. Dapps may begin using these methods immediately, falling back
+to `eth_sendTransaction`
+and `eth_getTransactionReceipt` when they are not available.
+
+### `wallet_sendMessageBundle`
+
+Requests that the wallet deliver a group of messages on-chain from the user's wallet.
+The wallet MAY send all the messages as a single transaction.
+Dapps MUST NOT rely on the messages being sent in a single transaction, i.e. other untrusted transactions may be
+included between each of the transactions.
+The wallet MUST send these messages in the order specified in the request.
+The wallet MUST attempt to deliver all messages if it returns a successful response to this method, and the wallet MUST
+NOT deliver any
+messages if it returns an error response.
+The wallet MAY reject the request if the chain ID does not match the currently selected chain ID.
+The wallet MUST send the messages on the specified chain ID.
+The wallet MUST reject the request if the `from` address does not match the enabled account.
+The wallet MAY reject the request if one or more messages in the bundle will fail.
+
+#### Parameters
+
+The method takes an array containing one object element, with the following keys:
+
+- `chainId`: `INTEGER` - The chain ID on which the transactions should be sent
+- `from`: `ADDRESS` - The address of the wallet that should send the messages in the bundle
+- `messages`: `ARRAY` - An array containing 1 or more message objects, where each message has the following keys:
+    - `to`: `ADDRESS` - (optional when creating new contract) The address the message is directed to.
+    - `gas`: `QUANTITY` - (optional) Integer of the gas provided for the message execution. Unused gas MAY be made
+      available to subsequent messages.
+    - `value`: `QUANTITY` - (optional) Integer of the value sent with this transaction.
+    - `data`: `DATA` - The compiled code of a contract OR the message data to include with the call, e.g. encoded
+      function signature and parameters.
+    - `optional`: `BOOLEAN` - (optional, default false) Whether the wallet should continue with subsequent messages if
+      this particular message fails
+
+##### Example
+
+```json
+[
+  {
+    "chainId": 1,
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "messages": [
+      {
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "gas": "0x76c0",
+        "value": "0x9184e72a",
+        "data": "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675",
+        "optional": true
+      },
+      {
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "gas": "0xdefa",
+        "value": "0x182183",
+        "data": "0xfbadbaf01"
+      }
+    ]
+  }
+]
+```
+
+#### Return value
+
+DATA, 32 Bytes - an identifier that the wallet uses to represent the bundle of messages. This identifier MUST persist
+between connections of the same Dapp and wallet.
```

Comment:
This discussion should happen on your `discussions-to` link, once you create one. In the meantime:

I don't like this design. It doesn't force interoperability between wallets. A bundle created with WalletA may or may not be usable with WalletB. I would suggest either:

```suggestion
DATA, 32 Bytes - an identifier that the wallet uses to represent the bundle of messages. This identifier MUST persist
between connections of the same Dapp and account.
```

Or allowing the dapp to specify the identifier.